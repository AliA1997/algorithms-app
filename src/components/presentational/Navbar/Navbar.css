.navbar {
    transition: position 2s;
    position: sticky;
    top: 0;
    grid-column: 1/4 !important;
    margin-left: auto;
    margin-right: auto;
    width: 100%;
    height: 50%;
    background: rgb(76, 156, 76);
    display: flex;
    justify-content: center;
    align-items: center;
    padding-top: 5%;
    padding-bottom: 5%;
}
.navbar-header {
    grid-row: 1/1;
    text-align: center;
    grid-column: 1/4;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 2em;
    font-weight: 700;
    height: 100%;
    margin: 0;
}
.desktop-nav {
    /* height: 100%; */
    display: none;
    /* grid-template-columns: 50% 20% 10% 10% 10%;
    display: grid; */
}
.mobile-nav-wrapper {
    grid-column: 1/4;
    width: 100%;
    height: 100%;
    text-align: center;
    display: grid;
    background: inherit;
    /* margin: 0; */
}
.mobile-nav {
    grid-column: 1/4;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    background: inherit;
}
.mobile-nav > * {
    display: flex;
    background: inherit;
    justify-content: center;
    align-items: center;
    padding: 5%;
    margin: 0;
    height: 100%;
}
.mobile-nav > p {
    font-size: 2em;
}
.mobile-nav > .navbar-img {
    display: flex;
    justify-content: center;
    height: 5em;
    width: 5em;
    border-radius: 50%;    
    margin-left: auto;
    margin-right: auto;
}

@media (min-width: 1200px) {
    .navbar > p:first-child {
        grid-column: 2/2;
    }
    .navbar > p:nth-child(2) {
        grid-column: 3/3;
    }
    .navbar > p:nth-child(3) {
        grid-column: 3/3;
    }
    .navbar > p:nth-child(4) {
        grid-column: 4/4;
    }
    .navbar > .navbar-img {
        grid-column: 5/5;
    }
}