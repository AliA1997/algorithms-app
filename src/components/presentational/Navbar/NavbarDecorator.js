//import withHandlers 
import { withHandlers } from 'recompose';

const decorator = withHandlers({
    onClick: ({linkFunc}) => (path) => () => {
        linkFunc(path);
    },
    onClick2: ({openMenu}) => e => {
        openMenu(e);
    }
})


//export decorator 
export default decorator;