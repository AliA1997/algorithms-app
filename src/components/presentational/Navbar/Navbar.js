import React from 'react';
// import your placeholder image 
import placeHolderImage from '../../../data/user-placeholder-image.jpg';
//import the decorator
import decorator from './NavbarDecorator';
import './Navbar.css';

export const Navbar = (props) => {
    const { user } = props;
    console.log('props----------', props.user)
    return (
        <div className='navbar'> 
            <div className='desktop-nav'>
                <p onClick={props.onClick('/')}>{`LearnCode</>`}</p>
                <p onClick={props.onClick('/topics')}>Topics</p>
                <p onClick={props.onClick('/blog')}>Blog</p>
                <img src='a' alt='a' className='navbar-img'/>
            </div>
            <div className='mobile-nav-wrapper'>
                <p onTouchMove={props.onClick('/')} onClick={props.onClick2} 
                className='navbar-header'>{`LearnCode</>`}</p>
                <div className='mobile-nav'  style={{display: props.open ? 'inline-block' : 'none'}}>
                    <p onClick={props.onClick('/topics')}>Topics</p>
                    <p onClick={props.onClick('/blog')}>Blog</p>
                    <img src={user && user.profile_picture || placeHolderImage} alt={user && user.username || 'placeholder image'}
                     className='navbar-img'/>
                </div>
            </div>
        </div>
    );
};

const NavbarDecorated = decorator(Navbar);
export default NavbarDecorated;