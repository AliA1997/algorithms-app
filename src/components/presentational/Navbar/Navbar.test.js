//import react and component
import React from 'react';
import { Navbar } from './Navbar';
//import render to create a mock component
import render from 'react-test-renderer';

//NOTE WHEN update snapshots use --updateSnapshot

//Check if the component renders correctly
describe('Check if component renders correctly', () => {
    //Check if the component renders correctly
    test('Check if the component renders correctly', () => {
        const tree = render.create(<Navbar />).toJSON();
        //Create a snapshot 
        expect(tree).toMatchSnapshot();
    })
})