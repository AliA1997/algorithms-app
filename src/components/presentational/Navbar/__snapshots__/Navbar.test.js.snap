// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if component renders correctly Check if the component renders correctly 1`] = `
<div>
  <div />
</div>
`;
